File: MagicHax/android/app/src/main/java/me/nathanp/magiccreatures/model/MediatorLiveFirebaseData.java
package me.nathanp.magiccreatures.model;

import com.google.firebase.database.ChildEventListener;
import com.google.firebase.database.DataSnapshot;
import com.google.firebase.database.DatabaseError;
import com.google.firebase.database.Query;

import java.util.List;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.lifecycle.MediatorLiveData;

public class MediatorLiveFirebaseData extends MediatorLiveData<DataSnapshot> {
    private Query query;
    private List<DataSnapshot> datas;
    private final LiveChildEventListener listener = new LiveChildEventListener();

    private class LiveChildEventListener implements ChildEventListener {

        @Override
        public void onChildAdded(@NonNull DataSnapshot dataSnapshot, @Nullable String s) {
            datas.add(dataSnapshot);
        }

        @Override
        public void onChildChanged(@NonNull DataSnapshot dataSnapshot, @Nullable String s) {

        }

        @Override
        public void onChildRemoved(@NonNull DataSnapshot dataSnapshot) {

        }

        @Override
        public void onChildMoved(@NonNull DataSnapshot dataSnapshot, @Nullable String s) {

        }

        @Override
        public void onCancelled(@NonNull DatabaseError databaseError) {

        }
    }
}

File: MagicHax/android/app/src/main/java/me/nathanp/magiccreatures/model/MediatorLiveFirebaseData.kt
package me.nathanp.magiccreatures.model

import androidx.lifecycle.MediatorLiveData
import com.google.firebase.database.ChildEventListener
import com.google.firebase.database.DataSnapshot
import com.google.firebase.database.DatabaseError
import com.google.firebase.database.Query

class MediatorLiveFirebaseData : MediatorLiveData<DataSnapshot?>() {
    private val query: Query? = null
    private val datas: MutableList<DataSnapshot>? = null
    private val listener = LiveChildEventListener()

    private inner class LiveChildEventListener : ChildEventListener {
        override fun onChildAdded(dataSnapshot: DataSnapshot, s: String?) {
            datas!!.add(dataSnapshot)
        }

        override fun onChildChanged(dataSnapshot: DataSnapshot, s: String?) {}
        override fun onChildRemoved(dataSnapshot: DataSnapshot) {}
        override fun onChildMoved(dataSnapshot: DataSnapshot, s: String?) {}
        override fun onCancelled(databaseError: DatabaseError) {}
    }
}